src/frontend/src/handler/cancel_job.rs

```diff
@@ -26,18 +26,51 @@ pub(super) async fn handle_cancel(
     jobs: JobIdents,
 ) -> Result<RwPgResponse> {
     let session = handler_args.session;
+    let job_ids = jobs.0;
+    let mut filtered_job_ids = vec![];
+    let mut notices = vec![];
+    {
+        let catalog_reader = session.env().catalog_reader().read_guard();
+        for job_id in job_ids {
+            let database_catalog = catalog_reader.get_database_by_name(&session.database())?;
+            let sink_catalog = database_catalog
+                .iter_schemas()
+                .find_map(|schema| schema.get_sink_by_id(&job_id));
+            if let Some(sink_catalog) = sink_catalog {
+                if sink_catalog.is_created() {
+                    continue; // Skip already created sinks
+                } else if sink_catalog.target_table.is_some() {
+                    notices.push(format!(
+                        "Please use `DROP SINK {}` to cancel sink into table job.",
```

Comment:
Will we continue to drop the sink? If not, perhaps we can just return an error here.

Reply:
Yes, the behavior of the `cancel` command is to accept multiple job parameters, and only the IDs that are successfully canceled will be returned. Therefore, I prefer not to return an error and instead add a notice, maintaining the behavior consistent with before.